nncf/experimental/torch/fracbits/quantizer.py

```diff
@@ -0,0 +1,222 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+# Reference: Yang, Linjie, and Qing Jin. "Fracbits: Mixed precision quantization via fractional bit-widths."
+# Proceedings of the AAAI Conference on Artificial Intelligence. Vol. 35. No. 12. 2021.
+
+from typing import Dict
+import torch
+
+from nncf.experimental.torch.fracbits.structs import FracBitsQuantizationMode
+
+from nncf.torch.layer_utils import COMPRESSION_MODULES, CompressionParameter
+from nncf.torch.quantization.layers import (
+    QUANTIZATION_MODULES, AsymmetricQuantizer, PTQuantizerSpec, SymmetricQuantizer)
+from nncf.torch.quantization.quantize_functions import asymmetric_quantize, symmetric_quantize
+from nncf.torch.utils import no_jit_trace
+
+
+@COMPRESSION_MODULES.register()
+@QUANTIZATION_MODULES.register(FracBitsQuantizationMode.SYMMETRIC)
+class FracBitsSymmetricQuantizer(SymmetricQuantizer):
+    def __init__(self, qspec: PTQuantizerSpec):
+        self._constructed = False
+        super().__init__(qspec)
+        self._min_num_bits = int(0.5 * qspec.num_bits)
+        self._max_num_bits = int(1.5 * qspec.num_bits)
```

Comment:
We can revisit min-max num bits setup mechanism after implementing `FracBitsCompressionBuilder` and `FracBitsCompressionController`. It can be up to `qspec` or calling `set_min_max_num_bits()` by `FracBitsCompressionBuilder` .